server/src/main/java/org/apache/iotdb/db/metadata/idtable/entry/DiskSchemaEntry.java

```diff
@@ -73,37 +77,61 @@ public DiskSchemaEntry(
   }
 
   public int serialize(OutputStream outputStream) throws IOException {
-    int byteLen = 0;
-    byteLen += ReadWriteIOUtils.write(deviceID, outputStream);
-    byteLen += ReadWriteIOUtils.write(seriesKey, outputStream);
-    byteLen += ReadWriteIOUtils.write(measurementName, outputStream);
-    byteLen += ReadWriteIOUtils.write(type, outputStream);
-    byteLen += ReadWriteIOUtils.write(encoding, outputStream);
-    byteLen += ReadWriteIOUtils.write(compressor, outputStream);
-    byteLen += ReadWriteIOUtils.write(isAligned, outputStream);
-
-    byteLen += ReadWriteIOUtils.write(byteLen, outputStream);
-    entrySize = byteLen;
-
-    return byteLen;
+    this.entrySize = ReadWriteIOUtils.write(toByteBuffer(), outputStream);
+    return entrySize;
```

Comment:
The byte data can be written directly into the ouputstream, which avoids the redundant data copy.

Reply:
```java
public ByteBuffer toByteBuffer() {
    byte[] deviceIDBytes = deviceID.getBytes();
    byte[] seriesKeyBytes = seriesKey.getBytes();
    byte[] measurementNameBytes = measurementName.getBytes();
    int length =
        deviceIDBytes.length
            + seriesKeyBytes.length
            + measurementNameBytes.length
            + Integer.BYTES * 3
            + Byte.BYTES * 3
            + ReadWriteIOUtils.BOOLEAN_LEN;
    ByteBuffer byteBuffer = ByteBuffer.allocate(length);
    byteBuffer.put(BytesUtils.intToBytes(deviceIDBytes.length));
    byteBuffer.put(deviceIDBytes);
    byteBuffer.put(BytesUtils.intToBytes(seriesKeyBytes.length));
    byteBuffer.put(seriesKeyBytes);
    byteBuffer.put(BytesUtils.intToBytes(measurementNameBytes.length));
    byteBuffer.put(measurementNameBytes);
    byteBuffer.put(type);
    byteBuffer.put(encoding);
    byteBuffer.put(compressor);
    ReadWriteIOUtils.write(isAligned, byteBuffer);
    return byteBuffer;
  }
```
Do you mean that the toByteBuffer() method has redundant data?

Reply:
I mean the process, that use a byteBuffer store the whole serialized binary data and then write it to outputStream, has some unnecessary operation. The data can be serialized to outputStream directly.

Reply:
fixed